src/Savegame/Country.cpp

```diff
@@ -333,4 +368,76 @@ bool Country::canBeInfiltrated()
 	return false;
 }
 
+namespace
+{
+
+int checkBoundsAndGet(int index, const std::vector<int>& element)
+{
+	if (index < 0 || index > element.size()) {
+		Log(LOG_WARNING) << "Attempted to get invalid data point.";
+		return -1;
+	}
+	return element[index];
+}
+
+}
+
+int Country::getMonthlyActivityAlien(int month) const { return checkBoundsAndGet(month, _activityAlien); }
```

Comment:
This function is not ergonomic for scripts, rewrite them to give "past" months from curret one, like:
```
foo(0) -> _activityAlien.back();
foo(1) -> _activityAlien[_activityAlien.size() - 2]
foo(2) -> _activityAlien[_activityAlien.size() - 3]
```
When you go over beginning you return `0` not `-1`.

Reply:
So you would just prefer this to increment backwards? I can do that, though my thinking was to provide an interface that could be easily looped over, in order to collect averages or sums or whatever over a range or portion of a range.

Reply:
But your version need to handle first 12 month in special way, you need multiple `if` to handle like less than 6 months.
If you use offset then you can always count last 6months same way even if you start game or if you are in 10y into game.
Another thing is that if we even expand length of tracked months then all mods will break if they assume 12 max.

Reply:
I guess I'm not following you. My thinking was I wanted to allow something like this to be easily possible:
```
var int months;
var int sum 0;

country.getMonthsTracked months;
# we want a 3 month average.
if months gt 3;
  set months 3;
end;

loop var month months;
  var int activity;
  country.getMonthlyActivityAlien activity;
  add sum activity;
end;

div sum months;
# do stuff with the average calculated.
```

I don't really follow the rest of your arguments, but I suppose it doesn't matter. I guess this can be done backward as easily as it is forwards, so I'll make the change.

Reply:
Compare this to:
```
var int m;
var int sum 0;

country.getMonthlyActivityAlien m 0;
add sum m;
country.getMonthlyActivityAlien m 1;
add sum m;
country.getMonthlyActivityAlien m 2;
add sum m;
div sum 3;
```
And this will work at day 1 and at day 10000, and your version need handling if you want get only last 3 months.
I think I would even add helper function that `add`:
```
var int sum 0;
country.addMonthlyActivityAlien sum 0;
country.addMonthlyActivityAlien sum 1;
country.addMonthlyActivityAlien sum 2;
div sum 3;
```
You could add loop too, but this would be cherry on top.

Reply:
Actually, this would fail on the first and second month, because, at that point there are only 1-2 months of data :P

But getting the months in forward or reverse order is a matter of little difference. I'll make the change.

Reply:
> Actually, this would fail on the first and second month, because, at that point there are only 1-2 months of data :P
> 
> But getting the months in forward or reverse order is a matter of little difference. I'll make the change.

How? whole point is that you can call `country.getMonthlyActivityAlien m 10000;` and it should work and set
value `0` to `m`.
This mean it will work for first game months too, of corse average will be 1/3 of only value, but this will be consistent with case when last year have no actions.

Reply:
Well, you have to make some decisions about what you want to do in the case when you want to average (say) 3 months of data, but there is only one or two months of data. But in most cases, I imagine using 0 for the missing value is probably not the behavior most frequently wanted. Instead, it generally makes more sense only to average the data present, which argues for returning some value that lets you distinguish between these two cases and/or telling the user that they did something somewhat nonsensical.

Reply:
I think I answer this in other post, to sum up, even if you get data from one month, you need pad values by appending "zeros" to make it behave similar to other month ranges. And if you really need make special handing for first months you can add `if` based on other function you already exposed.